android/vcmi-app/src/main/java/eu/vcmi/vcmi/ActivityLauncher.java

```diff
@@ -31,6 +32,7 @@ public void onCreate(@Nullable final Bundle savedInstanceState)
         super.onCreate(savedInstanceState);
         justLaunched = savedInstanceState == null;
         SDL.setContext(this);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
```

Comment:
it'd make more sense to enable it only when download starts and disable once it finishes

also, doesn't Qt or SDL already have such a function?

Reply:
No, I mean - what if player turns off screen during download? Won't it result in the same corruption?

Reply:
Can we at least detect such corruption & discard or restart download? Or this part is already working fine?

Reply:
That part should be fine then - mods are extracted into temporary directory, and then directory is moved to correct location. Such operation should be close to instant, or even atomic / single transaction

Reply:
> SDL is not relevant, since we only need this in Launcher.

SDL is linked to the launcher exe/lib, so it could be used (unless it requires initializing video system first).

> What about iOS?

iOS has the same issue. Once you turn off the screen or minimize an app, all sockets are closed.